### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,81 @@
...
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
 
-class DeliveryDrone {}
+    this.chipVersion = chipVersion;
```
Comment (andreyolin) [inline src/robotFactory.js:13]: you can move it to BaseRobot class constructor

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,81 @@
...
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
```
Comment (andreyolin) [inline src/robotFactory.js:45]: i think it`s not necessary to indicate x and y here, cause you indicated it in BaseRobot constructor

**Review by andreyolin — APPROVED** [review]
gj, my friend!

File: src/robotFactory.js (hunk 2 above)
Comment (Andriy-Shymkiv) [reply src/robotFactory.js:45]: i understood it when reviewing your code)

File: src/robotFactory.js (hunk 1 above)
Comment (Andriy-Shymkiv) [reply src/robotFactory.js:13]: but it is already there

**Review by denys-danyliuk — APPROVED** [review]

**Pull request closed without merging**